### PR TITLE
multiple day event

### PR DESCRIPTION
**Pull request by DoozyDoz**
This is in response to issue #1765

#### New Changes
<p float="left">
 					
  					 
  					
				   </p>

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/adapters/DayEventsAdapter.kt [hunk 1]
```diff
@@ -89,7 +91,16 @@ class DayEventsAdapter(activity: SimpleActivity, val events: ArrayList<Event>, r
             event_item_title.checkViewStrikeThrough(event.isTaskCompleted())
             event_item_time.text = if (event.getIsAllDay()) allDayString else Formatter.getTimeFromTS(context, event.startTS)
             if (event.startTS != event.endTS && !event.getIsAllDay()) {
-                event_item_time.text = "${event_item_time.text} - ${Formatter.getTimeFromTS(context, event.endTS)}"
+                val startDayCode = Formatter.getDayCodeFromTS(event.startTS)
+                val endDayCode = Formatter.getDayCodeFromTS(event.endTS)
+                val startTimePostFix = if (startDayCode != dayCode) Formatter.getDayTitle(activity, startDayCode,false) else null
+                val endTimePostFix = if (endDayCode != dayCode) Formatter.getDayTitle(activity, endDayCode,false) else null
+                val timeString = "${event_item_time.text}" +
+                    (if ( !startTimePostFix.isNullOrEmpty()) "(${startTimePostFix})" else "")  +
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calendar/pro/adapters/DayEventsAdapter.kt:99]: these lines are way too unreadable, please create local variables or something

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/adapters/DayEventsAdapter.kt [hunk 2]
```diff
@@ -93,14 +93,11 @@ class DayEventsAdapter(activity: SimpleActivity, val events: ArrayList<Event>, r
...
-                val startTimePostFix = if (startDayCode != dayCode) Formatter.getDayTitle(activity, startDayCode,false) else null
-                val endTimePostFix = if (endDayCode != dayCode) Formatter.getDayTitle(activity, endDayCode,false) else null
-                val timeString = "${event_item_time.text}" +
-                    (if ( !startTimePostFix.isNullOrEmpty()) "(${startTimePostFix})" else "")  +
-                    " - ${Formatter.getTimeFromTS(context, event.endTS)}"+
-                    (if ( !endTimePostFix.isNullOrEmpty()) "(${endTimePostFix})" else "")
-
-                event_item_time.text = timeString
+                val startTimeString = event_item_time.text
+                val startDayString = if (startDayCode != dayCode) "(${Formatter.getDayTitle(activity, startDayCode,false)})" else ""
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calendar/pro/adapters/DayEventsAdapter.kt:97]: theres still too much going on here

**Comment by DoozyDoz** [comment]
@tibbi did you look at the changes?

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/adapters/DayEventsAdapter.kt [hunk 3]
```diff
@@ -89,7 +91,15 @@ class DayEventsAdapter(activity: SimpleActivity, val events: ArrayList<Event>, r
             event_item_title.checkViewStrikeThrough(event.isTaskCompleted())
             event_item_time.text = if (event.getIsAllDay()) allDayString else Formatter.getTimeFromTS(context, event.startTS)
             if (event.startTS != event.endTS && !event.getIsAllDay()) {
-                event_item_time.text = "${event_item_time.text} - ${Formatter.getTimeFromTS(context, event.endTS)}"
+                val startDayCode = Formatter.getDayCodeFromTS(event.startTS)
+                val endDayCode = Formatter.getDayCodeFromTS(event.endTS)
+                val startDate = Formatter.getDayTitle(activity, startDayCode,false)
+                val endDate = Formatter.getDayTitle(activity, endDayCode,false)
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calendar/pro/adapters/DayEventsAdapter.kt:97]: format the code with ctrl + shift + L, there are some missing spaces in both files.

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/adapters/DayEventsAdapter.kt [hunk 4]
```diff
@@ -41,7 +43,7 @@ class DayEventsAdapter(activity: SimpleActivity, val events: ArrayList<Event>, r
 
     override fun actionItemPressed(id: Int) {
         when (id) {
-            R.id.cab_share -> shareEvents()
+            R.id.cab_share  -> shareEvents()
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calendar/pro/adapters/DayEventsAdapter.kt:46]: double space here

**Comment by tibbi** [comment]
should be good, thanks

**Pull request merged**